src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -703,6 +706,163 @@ def pre_instance_update(self, instance):
+            for plc in instance.properties.private_link_configurations:
+                if plc.name == args.name:
+                    err_msg = "Private link name duplicates."
+                    raise ValidationError(err_msg)
+        # prepare subnet for new private link
+        from .aaz.latest.network.vnet import Show
+        rid = instance.properties.gateway_ip_configurations[0].properties.subnet.id.to_serialized_data()
+        vnet_name = parse_resource_id(rid)["name"]
+        vnet = Show(cli_ctx=self.cli_ctx)(command_args={
+            "name": vnet_name,
+            "resource_group": args.resource_group
+        })
+        for subnet in vnet["subnets"]:
+            if subnet["name"] == args.subnet:
+                err_msg = "Subnet name duplicates. In order to use existing subnet, please enter subnet ID."
+                raise ValidationError(err_msg)
+            cond1 = subnet["addressPrefix"] == args.subnet_prefix
+            cond2 = "addressPrefixes" in subnet and args.subnet_prefix in subnet["addressPrefixes"]
+            if cond1 or cond2:
+                err_msg = "Subnet prefix duplicates."
+                raise ValidationError(err_msg)
+
+        if is_valid_resource_id(args.subnet.to_serialized_data()):
+            subnet_id = args.subnet
+        else:
+            subnet_id = resource_id(
+                subscription=self.ctx.subscription_id,
+                resource_group=args.resource_group,
+                namespace="Microsoft.Network",
+                type="virtualNetworks",
+                name=vnet_name,
+                child_type_1="subnets",
+                child_name_1=args.subnet
+            )
+
+            from azure.cli.core.commands import LongRunningOperation
+            poller = VNetSubnetCreate(cli_ctx=self.cli_ctx)(command_args={
+                "name": args.subnet,
+                "vnet_name": vnet_name,
+                "address_prefix": args.subnet_prefix,
+                "resource_group": args.resource_group,
+                "private_link_service_network_policies": "Disabled"
+            })
+            LongRunningOperation(self.cli_ctx)(poller)
+
+        args.ip_configurations = [{
+            "name": "PrivateLinkDefaultIPConfiguration",
+            "private_ip_address": args.ip_address,
+            "private_ip_allocation_method": "Static" if has_value(args.ip_address) else "Dynamic",
+            "subnet": {"id": subnet_id},
+            "primary": args.primary
+        }]
+
+
+class AGPrivateLinkRemove(_AGPrivateLinkRemove):
+    def pre_instance_delete(self):
+        args = self.ctx.args
+        instance = self.ctx.vars.instance
+        for plc in instance.properties.private_link_configurations:
+            if plc.name == args.name:
+                to_be_removed = plc
+                break
+        else:
+            err_msg = "Private link doesn't exist."
+            raise ValidationError(err_msg)
```

Comment:
Does old version of python version support `for...else...`?

Reply:
It has been introduced from 1990...